llama_stack/providers/adapters/memory/qdrant/qdrant.py

```diff
@@ -0,0 +1,167 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the terms described in the LICENSE file in
+# the root directory of this source tree.
+
+import traceback
+import uuid
+from typing import Any, Dict, List
+
+from numpy.typing import NDArray
+from qdrant_client import AsyncQdrantClient, models
+from qdrant_client.models import PointStruct
+
+from llama_stack.providers.datatypes import MemoryBanksProtocolPrivate
+
+from llama_stack.apis.memory import *  # noqa: F403
+
+from llama_stack.providers.adapters.memory.qdrant.config import QdrantConfig
+from llama_stack.providers.utils.memory.vector_store import (
+    BankWithIndex,
+    EmbeddingIndex,
+)
+
+CHUNK_ID_KEY = "_chunk_id"
+
+
+def convert_id(_id: str) -> str:
+    """
+    Converts any string into a UUID string based on a seed.
+
+    Qdrant accepts UUID strings and unsigned integers as point ID.
+    We use a seed to convert each string into a UUID string deterministically.
+    This allows us to overwrite the same point with the original ID.
+    """
+    return str(uuid.uuid5(uuid.NAMESPACE_DNS, _id))
+
+
+class QdrantIndex(EmbeddingIndex):
+    def __init__(self, client: AsyncQdrantClient, collection_name: str):
+        self.client = client
+        self.collection_name = collection_name
+
+    async def add_chunks(self, chunks: List[Chunk], embeddings: NDArray):
+        assert len(chunks) == len(
+            embeddings
+        ), f"Chunk length {len(chunks)} does not match embedding length {len(embeddings)}"
+
+        if not await self.client.collection_exists(self.collection_name):
+            await self.client.create_collection(
```

Comment:
can we move this into `register_memory_bank` instead? I think we are going to make it so that the source of truth about this memory bank stays within the provider itself -- so Qdrant should know about this already by the time a query comes in. then, inside `list_memory_banks` you can actually query Qdrant server to find all relevant collections associated with the Stack.

Reply:
Qdrant needs the vector embeddings dimensions when creating a collection. It's available here. But not in `register_memory_bank`

Also, Qdrant doesn't include support collection-level metadata to store bank info.

Reply:
OK thanks -- can we move the creation of the collection into the `register_memory_bank` though? by the time we reach this place (adding chunks to a collection), it should have already existed otherwise there's some issue.